### PR TITLE
set up association

### PR DESCRIPTION
**Pull request by Yemisirach**
- Create a class `Classroom` with the following:
  - `@label` instance variable, should be initialized in the constructor.
  - Setter and getter for `@label` (remember about `attr_accessor`).
- Create the `has-many`/`belongs-to` relationship between `Classroom` and `Student`. The following should be implemented:
  - Create the `has-many` side (a classroom has many students).
  - Create the `belongs-to` side (a student belongs to a classroom).
  - Make sure that when adding a student to a classroom it also sets the classroom for the student.
  - Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
- Create a class `Book` with the following:
  - `@title` and `@author` instance variables, should be initialized in the constructor.
  - Setters and getters for instance variables (remember about `attr_accessor`).
- Create a class `Rental` with the following:
  - `@date` instance variable, should be initialized in the constructor.
  - Setter and getter for `@date` (remember about `attr_accessor`).
- Create the `many-to-many` (also `has-many-through`) relationship between `Person` and `Book` using the intermediate class `Rental`. The following should be implemented:
  - Create the `has-many` side of `Book` and `Rental` (a book has many rentals).
  - Create the `belongs-to` side of `Rental` and `Book` (a rental belongs to a book).
  - Create the `has-many` side of `Person` and `Rental` (a person has many rentals).
  - Create the `belongs-to` side of `Rental` and `Person` (a rental belongs to a person).
  - Modify the constructor of `Rental` so `Book` and `Person` are set in it.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (m05e5) [inline book.rb]: - [ ] Well done with your Book class. As part of the requirements, you are supposed to create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. So, Kindy Implements the has-many relation in Book by creating a method to add rentals in the Book class.

File: person.rb [hunk 2]
```diff
@@ -10,6 +10,7 @@ def initialize(age, name: 'Unknown', parent_permission: true)
     @name = name,
             @age = age,
             @parent_permission = parent_permission
+    @rentals = []
```
Comment (m05e5) [inline person.rb:16]: - [ ] Well done with your Person class. As part of the requirements, you are supposed to create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. So, Kindy Implements the has-many relation in Person by creating a method to add rentals in the Person class.

**Review by m05e5 — CHANGES_REQUESTED** [review]
Hi @Yemisirach , 

Good job so far. !
There are some issues that you still need to work on to go to the next project but you are almost there!
mario_ladders


## Highlights


## Required Changes ♻️
* You really did a great job but there are some features you need to improve

- [ ] Please check the comments below 

_I'm sure you can do it just keep it up_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_@m05e5_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (rloterh) [inline book.rb:9]: - [ ] Like informed in the previous review, a _rental_ can have many _books_, and _books_ can be subjected to many _rentals_. In order for us to establish a _many-to-many_ association between _rentals_ and _books_ (let's bear in mind that the association between _person_ and _rentals_ is also _many-to-many_), let's define a method for adding rentals within the book object similar to the example below:

```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .

  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: book.rb (hunk 1 above)
Comment (rloterh) [reply book.rb]: Needs attention!

File: person.rb (hunk 2 above)
Comment (rloterh) [reply person.rb:16]: well resolved! ✔️

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Yemi, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🚦 _ 🏡 🏃‍♂️ 🏚️ 🌴 🗼

## Highlights 👍  
- Your code OOP is taking shape!

## Required Changes ♻️
_Kindly check the inline comments above/under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

**You can also consider:**
- In addition, kindly ensure to add the _Future Features_ section to your ReadMe, which should include 1-3 potential features to be added.
Implementing the above change(s) will make your ReadMe/repository stand out even better! 😉

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi @Yemisirach , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Good class relationship setup :heavy_check_mark: 
- No rubocop errors :heavy_check_mark: 
- Good use of gitflow :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**